specs/core/1_shard-data-chains.md

```diff
@@ -75,207 +91,191 @@ The following types are defined, mapping into `DomainType` (little endian):
+                         shard: Shard) -> List[ValidatorIndex, MAX_PERSISTENT_COMMITTEE_SIZE]:
     """
     Return committee for a period. Used to construct persistent committees.
     """
-    return compute_committee(
+    full_committee = compute_committee(
         indices=get_active_validator_indices(state, epoch),
         seed=get_seed(state, epoch),
         index=shard * count + index,
         count=SHARD_COUNT * count,
     )
-```
 
-### `get_switchover_epoch`
-
-```python
-def get_switchover_epoch(state: BeaconState, epoch: Epoch, index: ValidatorIndex) -> int:
-    earlier_start_epoch = Epoch(epoch - (epoch % PERSISTENT_COMMITTEE_PERIOD) - PERSISTENT_COMMITTEE_PERIOD * 2)
-    return (bytes_to_int(hash(get_seed(state, earlier_start_epoch) + int_to_bytes(index, length=3)[0:8]))
-            % PERSISTENT_COMMITTEE_PERIOD)
+    return [i for i in full_committee if is_active_validator(state.validators[i], epoch)][:MAX_PERSISTENT_COMMITTEE_SIZE]
 ```
 
 ### `get_persistent_committee`
 
 ```python
 def get_persistent_committee(state: BeaconState,
                              shard: Shard,
-                             slot: Slot) -> Sequence[ValidatorIndex]:
+                             slot: ShardSlot) -> Sequence[ValidatorIndex]:
     """
     Return the persistent committee for the given ``shard`` at the given ``slot``.
     """
-    epoch = compute_epoch_of_slot(slot)
-    earlier_start_epoch = Epoch(epoch - (epoch % PERSISTENT_COMMITTEE_PERIOD) - PERSISTENT_COMMITTEE_PERIOD * 2)
-    later_start_epoch = Epoch(epoch - (epoch % PERSISTENT_COMMITTEE_PERIOD) - PERSISTENT_COMMITTEE_PERIOD)
+    epoch = compute_epoch_of_shard_slot(slot)
 
-    committee_count = max(
-        len(get_active_validator_indices(state, earlier_start_epoch)) //
-        (SHARD_COUNT * TARGET_COMMITTEE_SIZE),
-        len(get_active_validator_indices(state, later_start_epoch)) //
-        (SHARD_COUNT * TARGET_COMMITTEE_SIZE),
-    ) + 1
-
-    index = slot % committee_count
-    earlier_committee = get_period_committee(state, earlier_start_epoch, shard, index, committee_count)
-    later_committee = get_period_committee(state, later_start_epoch, shard, index, committee_count)
+    earlier_committee = get_period_committee(state, get_period_start_epoch(epoch, lookback=2), shard)
+    later_committee = get_period_committee(state, get_period_start_epoch(epoch, lookback=1), shard)
 
     # Take not-yet-cycled-out validators from earlier committee and already-cycled-in validators from
     # later committee; return a sorted list of the union of the two, deduplicated
-    return sorted(list(set(
-        [i for i in earlier_committee if epoch % PERSISTENT_COMMITTEE_PERIOD < get_switchover_epoch(state, epoch, i)]
-        + [i for i in later_committee if epoch % PERSISTENT_COMMITTEE_PERIOD >= get_switchover_epoch(state, epoch, i)]
-    )))
+    return sorted(set(
+        [i for i in earlier_committee if epoch % EPOCHS_PER_SHARD_PERIOD < i % EPOCHS_PER_SHARD_PERIOD]
```

Comment:
The point is specifically to use the validator index. It mirrors how staggering currently works for the proof of custody, where transition epoch for a validator equals validator index mod committee length.

Reply:
Doesn't it make more since to take a simple slice of each to ensure that we actually have the property of committee being less than or equal to `MAX_PERSISTENT_COMMITTEE_SIZE`?

Reply:
I see now. We can't have a simple 1-in/1-out from the `earlier` and `later` committee because we might have someone assigned to multiple committees at once. Instead need a private switchover epoch per validator index.

Thinking about alternatives..